homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -27,46 +31,49 @@
 )
 from .const import DOMAIN  # pylint: disable=unused-import
 
+_LOGGER = logging.getLogger(__name__)
+
+
+def _schema_with_defaults(user_input=None):
+    if user_input is None:
+        user_input = {}
+
+    return vol.Schema(
+        {
+            vol.Optional(
+                CONF_NAME, default=user_input.get(CONF_NAME, DEFAULT_NAME)
+            ): str,
+            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+            vol.Optional(CONF_PORT, default=user_input.get(CONF_PORT, "")): str,
+            vol.Optional(CONF_SSL, default=user_input.get(CONF_SSL, DEFAULT_SSL)): bool,
+            vol.Optional(
+                CONF_API_VERSION,
+                default=user_input.get(CONF_API_VERSION, DEFAULT_DSM_VERSION),
+            ): vol.All(
+                vol.Coerce(int),
+                vol.In([5, 6]),  # DSM versions supported by the library
+            ),
```

Comment:
Is this something we can detect? Seems odd to ask the user to figure this out.

Reply:
@Quentame Can detection added to the pypi package?

Reply:
Not for now, but will look at it for the next version of the lib.
Can be done as it is for now, and will update when available.

Actually, there is some kind of trick we can do but I don't like that :
There is a `dsm_api.information.version_string` property, but the endpoint already depends on the API version, see ProtoThis/python-synology#13
So we can try to reach it with one API version + test with the other if not ... :hankey: 

I would prefer handling the autodetect API version directly inside the lib so we can remove this param.

The v5 support was added recently with ProtoThis/python-synology#9 by @aaska which needed it for HA.

Will ask @aaska to help me implementing this.

Just created an "issue" to help maintainers of the lib.
Synology NAS owner, pleaser register here  ProtoThis/python-synology#26 !